src/test/initialize.ts

```diff
@@ -76,23 +76,43 @@ export async function closeActiveNotebooks(): Promise<void> {
 }
 
 async function closeWindowsInternal() {
-    return new Promise<void>((resolve, reject) => {
-        // Attempt to fix #1301.
-        // Lets not waste too much time.
-        const timer = setTimeout(() => {
-            reject(new Error("Command 'workbench.action.closeAllEditors' timed out"));
-        }, 15000);
-        vscode.commands.executeCommand('workbench.action.closeAllEditors').then(
-            () => {
-                clearTimeout(timer);
-                resolve();
-            },
-            (ex) => {
-                clearTimeout(timer);
-                reject(ex);
-            }
-        );
-    });
+    class CloseEditorsTimeoutError extends Error {
+        constructor() {
+            super("Command 'workbench.action.closeAllEditors' timed out");
+        }
+    }
+    const closeWindowsImplementation = (timeout = 15_000) => {
```

Comment:
Found in the PR that VS Code was timing out here.
We've had this issue in the past hence the timeout.
Updated to ensure we retry again after sleeping for 1/2s.
Possible VSC is stuck/busy, hence giving it time & retrying.

I.e. a work around for something flaky in VS Code.